mdp/nodes/sfa_nodes.py

```diff
@@ -369,6 +378,200 @@ def get_eta_values(self, t=1):
+            which would save a potentially time-consuming rerun of all
+            ``train()`` calls.
+        :type rank_deficit_method: str
+        """
+        super(VartimeSFANode, self).__init__(
+                input_dim, output_dim, dtype, include_last_sample=True,
+                rank_deficit_method=rank_deficit_method)
+
+        self.tchunk = 0
+
+    def _init_cov(self):
+        # init two covariance matrices
+        # one for the input data
+        self._cov_mtx = VartimeCovarianceMatrix(self.dtype)
+        # one for the derivatives
+        self._dcov_mtx = VartimeCovarianceMatrix(self.dtype)
+
+    def time_derivative(self, x, dt=None):
+        """
+        Compute the linear approximation of the time derivative
+
+        :param x: The time series data.
+        :type x: numpy.ndarray
+
+        :param dt: Sequence of time increments between vectors. 
+
+            Usage with only single chunk of data:
+                *dt* must be of length *x.shape[0]-1*.
+
+            Usage with multiple chunks of data with intended time dependence:
+                *dt* should have length *x.shape[0]-1* in the first call and
+                be of length *x.shape[0]* starting with the second call.
+                Starting with the second call, the first element in each chunk
+                will be considered the time difference between the last element
+                of *x* in the previous chunk and the first element of *x* of the
+                current chunk.
+
+            Usage with multiple chunks without time dependence:
+                If *dt* has length *x.shape[0]-1* time dependence between
+                chunks is omitted and all algorithmic components regard
+                only the time structure within chunks.
+
+            Minimal usage:
+                If *dt* is omitted entirely there will be no time dependence
+                between chunks and it will be considered to be one everywhere
+                within a chunk.
+
+        :type dt: numpy.ndarray
+
+        :returns: Piecewise linear approximation of the time derivative.
+        :rtype: numpy.ndarray
+        """
+        outlen = x.shape[0] - 1\
+                if self.tchunk == 0 or dt is None or dt.shape[0] != x.shape[0] else x.shape[0]
+        out = numx.empty([outlen, x.shape[1]])
+
+        if dt is not None:
+            # Improvements can be made, by interpolating polynomials
+            out[-x.shape[0]+1:, :] =\
+                    (x[1:, :]-x[:-1, :]) / dt[-x.shape[0]+1:, None]
+        else:
+            # trivial fallback
+            out[-x.shape[0]+1:, :] = x[1:, :]-x[:-1, :]
+
+        if self.tchunk > 0 and dt is not None and dt.shape[0] == x.shape[0]:
+            # check if str before checking, to avoid comparison warnings
+            out[0, :] = numx.divide(x[0, :]-self.xlast, dt[0])
+
+        self.xlast = x[-1, :].copy()
+        self.dtlast = dt[-1].copy() if dt is not None else 1.
```

Comment:
`self.dtlast` seems to be never used ecxept in this line. Why do we store it then?

Reply:
I hope we won't need it again for some reason.